build.gradle.kts

```diff
@@ -9,6 +14,75 @@ allprojects {
     group = findProperty("GROUP").toString()
 }
 
+subprojects {
+    tasks.withType<JavaCompile>().configureEach {
+        sourceCompatibility = JavaVersion.VERSION_17.toString()
+        targetCompatibility = JavaVersion.VERSION_17.toString()
+    }
+
+    tasks.withType<KotlinCompile>().configureEach {
+        kotlinOptions {
+            jvmTarget = JavaVersion.VERSION_17.toString()
+        }
+    }
+
+    plugins.withType<BasePlugin>().configureEach {
+        extensions.configure<BaseExtension> {
+            compileSdkVersion(libs.versions.android.compile.sdk.get().toInt())
+            buildToolsVersion(libs.versions.android.build.tools.get())
+            ndkVersion = libs.versions.android.ndk.get()
+
+            defaultConfig {
+                minSdk = libs.versions.android.min.sdk.get().toInt()
+                targetSdk = libs.versions.android.target.sdk.get().toInt()
+            }
+
+            compileOptions {
+                sourceCompatibility = JavaVersion.VERSION_17
+                targetCompatibility = JavaVersion.VERSION_17
+            }
+        }
+
+        extensions.findByType<LibraryExtension>()?.apply {
```

Comment:
For `LibraryExtension`s only (so not for `ApplicationPlugin` modules), then apply this config.